walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,51 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> times = new ArrayList<>();
+    Collection<String> attendees = request.getAttendees();
+    long duration = (long) request.getDuration();
+    if (duration > TimeRange.WHOLE_DAY.duration()){
+      return times;
+    }
+    times.add(TimeRange.WHOLE_DAY);
+    if (attendees.isEmpty()){
+      return times;
+    }
+    Collection<TimeRange> tempTimes;
+    for (Event event: events){
+      for (String attendee: attendees){
+        if (event.getAttendees().contains(attendee)){
+          tempTimes = new ArrayList<>();
```

Comment:
nit: naming.  I'd suggest avoiding the word "temp" when naming variables since it does not clearly describe what this variable contains (or why it's a temporary list in the first place).  Maybe "times_for_validation" or "pending_validation" or something similar?